src/ucp/core/ucp_worker.c

```diff
@@ -1181,13 +1181,21 @@ static void ucp_worker_close_ifaces(ucp_worker_h worker)
     UCS_ASYNC_UNBLOCK(&worker->async);
 }
 
-static void
-ucp_worker_get_sys_device_distance(ucp_context_h context,
-                                   ucp_rsc_index_t rsc_index,
-                                   ucs_sys_dev_distance_t *distance)
+static ucs_sys_device_t
+ucp_worker_iface_get_sys_device(const ucp_worker_iface_t *wiface)
+{
+    const ucp_context_h context = wiface->worker->context;
+    ucp_rsc_index_t rsc_index   = wiface->rsc_index;
+
+    return context->tl_rscs[rsc_index].tl_rsc.sys_device;
+}
+
+static void ucp_worker_iface_get_sys_device_distance(ucp_worker_iface_t *wiface)
```

Comment:
`const`?

Reply:
func name is misleading... this function actually change the state of the wiface.
I will change it to `set`